source/common/upstream/upstream_impl.cc

```diff
@@ -506,9 +506,18 @@ Host::CreateConnectionData HostImpl::createConnection(
         socket_factory.createTransportSocket(transport_socket_options, host),
         upstream_local_address.socket_options_, transport_socket_options);
   } else if (address_list.size() > 1) {
-    connection = std::make_unique<Network::HappyEyeballsConnectionImpl>(
-        dispatcher, address_list, source_address_selector, socket_factory, transport_socket_options,
-        host, options);
+    if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.use_config_in_happy_eyeballs") &&
+        cluster.happyEyeballsConfig().has_value()) {
+      ENVOY_LOG(debug, "Upstream using happy eyeballs config.");
+      connection = std::make_unique<Network::HappyEyeballsConnectionImpl>(
+          dispatcher, address_list, source_address_selector, socket_factory,
+          transport_socket_options, host, options, cluster.happyEyeballsConfig());
```

Comment:
Nit: Let's simplify this by doing something like:
```
optional<HappyEyeballsConfig> config;
if (Runtime::runtimeFeatureEnabled(...) {
  config = cluster.happyEyeballsConfig();
}
connection = std::make_unique<Network::HappyEyeballsConnectionImpl>(
          dispatcher, address_list, source_address_selector, socket_factory,
          transport_socket_options, host, options, config);
```
(or some such)

Reply:
done